frontend/src/host_orchestrator/orchestrator/instancemanager.go

```diff
@@ -343,60 +366,50 @@ func (m *CVDToolInstanceManager) launchFromAndroidCI(
 	if merr != nil {
 		return nil, merr
 	}
-	instanceNumber := atomic.AddUint32(&m.instanceCounter, 1)
-	cvdName := fmt.Sprintf("cvd-%d", instanceNumber)
-	runtimeDir := m.paths.RuntimesRootDir + "/" + cvdName
-	if err := createNewDir(runtimeDir); err != nil {
-		return nil, err
-	}
 	startParams := startCVDParams{
-		InstanceNumber:   instanceNumber,
+		InstanceNumbers:  m.newInstanceNumbers(1 + req.AdditionalInstancesNum),
```

Comment:
nit: `+1` is error prone, so it should be done as few times as necessary, yet it is done in two different functions. Given that both `launchFromAndroidCI` and `launchFromUserBuild` only access `req.BuildSource` and `req.AdditionalIntancesNum` it would be better to pass to both functions `numInstances` and `buildSource` and compute `numInstances` only once at the call site.

Reply:
agree, done